### ISSUE TITLE
Plot not showing up

### ISSUE THREAD

**Issue by sciadopitys** [davidtvs/pytorch-lr-finder#72]
I'm seeing a suggested learning rate, but no plot when calling .plot() with all default arguments.

**Comment by NaleRaphael** [comment]
Hi @sciadopitys,

The notebook `examples/lrfinder_mnist.ipynb` can be executed successfully without problems, maybe you can try from it first.
Besides, could you attach the script of how you run with `LRFinder`? If there is any error message, it would be better to post them here too.

**Comment by sciadopitys** [comment]
Hello,
I'm actually running my script on a remote server.  I simply copy-and-pasted the lines from the gradient accumulation section:

desired_batch_size, real_batch_size = 44, 2
accumulation_steps = desired_batch_size // real_batch_size
lr_finder = LRFinder(model, optimizer, criterion, device="cuda")
lr_finder.range_test(train_loader, end_lr=.001, num_iter=100, step_mode="exp", accumulation_steps=accumulation_steps)
lr_finder.plot()
lr_finder.reset()

, only changing the batch sizes and the end_lr argument of range_test.  The lr finder seems to be working correctly and no error message is produced, its just that no plot is displayed at the end.  All I see is

Learning rate search finished. See the graph with {finder_name}.plot()
LR suggestion: steepest gradient
Suggested LR: value

**Comment by davidtvs** [comment]
If you are running in a remote server matplotlib probably can't create the graph because the display backend is not set: https://stackoverflow.com/questions/3453188/matplotlib-display-plot-on-a-remote-machine

**Comment by NaleRaphael** [comment]
OK, I'm not sure whether the actual problem is currently.

If I understand you correctly, the title "Plot not showing up" means "the window of a `matplotlib` plot didn't show up", rather than "content of a `matplotlib` plot window is empty".

Regarding this meaning, my first guess it's resulted by the environment you were using.
If you run the script through a terminal + SSH on remote server, there might be extra setup you have to done to make the plot able to show up on your client machine. Or if you run the script in a Jupyter Notebook, we requires more information to investigate the issue.

But here is a thing you can try first: could you check that `lr_finder.history` isn' empty after running `lr_finder.range_test()`?

**Comment by sciadopitys** [comment]
It's definitely an issue with the display backend.  However, I'm still getting "gdk_cursor_new_for_display: assertion 'GDK_IS_DISPLAY (display)' failed" even when I specify matplotlib to use a different backend.  Regardless, the issue doesn't really have to do with pytorch-lr-finder itself.

**Comment by NaleRaphael** [comment]
If it still doesn't work after configuring SSH config (X11 forwarding), maybe you can try the following workarounds:
1. Save the figure as an image, then copy it and view it on your local machine. (see also [this comment](https://github.com/davidtvs/pytorch-lr-finder/pull/23#issuecomment-602248697))
2. Save the figure as an image, then view it with other tools. e.g. https://superuser.com/questions/557622
3. Try [`matplotlib-terminal`](https://github.com/matrach/matplotlib-terminal).
4. Use `Jupyter notebook` or `JupyterLab`.
